python/GafferUI/NodeSetEditor.py

```diff
@@ -133,19 +133,23 @@ def setNodeSetDriver( self, drivingEditor, mode = DriverModeNodeSet ) :
 
 			drivingEditor.__registerDrivenEditor( self, mode )
 
-			weakDriver = weakref.ref(
-				drivingEditor,
-				# We need to unlink ourselves if the driver goes away
-				lambda _ : self.setNodeSetDriver( None )
-			)
+			weakSelf = weakref.ref( self )
+
+			# We need to unlink ourselves if the driver goes away
+			def disconnect( _ ) :
+				if weakSelf() is not None:
+					weakSelf().setNodeSetDriver( None )
+
+			weakDriver = weakref.ref( drivingEditor, disconnect )
+
 			changeCallback = self.__nodeSetDriverModes[ mode ]
 
 			def updateFromDriver( _ ) :
 				if weakDriver() is not None :
 					nodeSet = weakDriver().getNodeSet()
 					if changeCallback :
-						nodeSet = changeCallback( self, weakDriver() )
-					self.__setNodeSetInternal( nodeSet, callUpdateFromSet=True )
+						nodeSet = changeCallback( weakSelf(), weakDriver() )
+					weakSelf().__setNodeSetInternal( nodeSet, callUpdateFromSet=True )
```

Comment:
Do we know that `weakSelf()` isn't `None` at this point? I'm not familiar enough with the code to easily determine that. Also, can we expose the original bug via a unit test? We've had a couple of these editor-circular-reference bugs of late, so any extra protection we can get would be valuable.

Reply:
It shouldn't be, but I've fixed it in https://github.com/GafferHQ/gaffer/pull/3421/commits/61d14fc90b0677d41b1258b8d0b145ee19e38bb9 either way to be safe.

Reply:
Tests in https://github.com/GafferHQ/gaffer/pull/3421/commits/61d14fc90b0677d41b1258b8d0b145ee19e38bb9 too.